src/read.jl

```diff
@@ -261,3 +255,225 @@ function read_demand_flexibility(filepath, interval)::DemandFlexibility
+                        if any(x -> x < 0, fh[:, zone_columns_idx[flexible_zone_num] + 1])
+                            @error(
+                                "Input ERROR: Total zone flexibility less than sum of bus flexibility for zone " *
+                                    string(flexible_zone_id[i])
+                            )
+                            throw(ErrorException("See above."))
+                        end
+                    end
+                end
+
+                # convert zone-level to bus-level
+                converted_zone_flexibility =
+                    Matrix(fh[:, [i + 1 for i in zone_columns_idx]]) * zone_to_bus_shares[zone_cols_idx, :]
+
+                # add bus-level columns on top of converted bus-level flexibility matrix
+                flex_bus_idx = zeros(Int64, 0)
+                for i in sets.bus_idx
+                    bus = case.busid[i]
+                    # add specified flexibility to the corresponding bus
+                    if bus in flexible_bus_id
+                        converted_zone_flexibility[:, i] += fh[string(bus)]
+                    end
+
+                    # identify flexible bus by their total flexibility and append to list
+                    if any(x -> x > 0, converted_zone_flexibility[:, i])
+                        append!(flex_bus_idx, i)
+                    end
+                end
+                eq_bus_df = DataFrames.DataFrame(
+                    converted_zone_flexibility[:, flex_bus_idx]
+                )
+                # for cost, the zone numbers apply to all buses except those with dedicated columns
+            else
+                # convert zone-level cost using incidence matrix
+                converted_zone_cost =
+                    Matrix(fh[:, [i + 1 for i in zone_columns_idx]]) * zone_to_bus_incidence[zone_cols_idx, :]
+
+                # replace bus-level cost for bus columns in converted bus-level cost matrix
+                flex_bus_idx = zeros(Int64, 0)
+                for i in sets.bus_idx
+                    bus = case.busid[i]
+                    # add specified cost to the corresponding bus
+                    if bus in flexible_bus_id
+                        converted_zone_cost[:, i] = fh[string(bus)]
+                    end
+
+                    # identify flexible bus by their total cost and append to list
+                    if any(x -> x > 0, converted_zone_cost[:, i])
+                        append!(flex_bus_idx, i)
+                    end
+                end
+                eq_bus_df = DataFrames.DataFrame(converted_zone_cost[:, flex_bus_idx])
+            end
+
+            # add header and datetime index column
+            DataFrames.rename!(eq_bus_df, Symbol.(case.busid[flex_bus_idx]))
+            DataFrames.insertcols!(eq_bus_df, 1, :"UTC Time" => fh["UTC Time"])
+
+            # store to new df object
+            demand_flexibility_updated[field] = eq_bus_df
```

Comment:
Have you come across a problem with adding DataFrames to the `demand_flexibility_updated` dictionary during your tests? In testing this function, I've run into problems where the initialized dictionary is created to be of type `Dict{String, Union{Nothing, Integer}}` (i.e., a dictionary with keys that are strings and values that are either nothing or integers). Since this is how the dictionary is initialized, I'm not able to add a DataFrame to the dictionary, either by adding a new key or by updating an already established key. I'm not sure if this is due to a difference in Julia versions or package versions, so I want to check. If this is a common problem, it appears that the work around is by specifying that the dictionary can accept keys that are strings and values that can be any datatype through the following:

```julia
demand_flexibility_updated = Dict{String, Any}(
    "duration" => demand_flexibility.duration,
    "enabled" => demand_flexibility.enabled,
    "interval_balance" => demand_flexibility.interval_balance,
    "rolling_balance" => demand_flexibility.rolling_balance,
    "flex_amt_up" => nothing,
    "flex_amt_dn" => nothing,
    "cost_up" => nothing,
    "cost_dn" => nothing,
)
```

Reply:
what is the version of Julia that you are using? I have Julia 1.5.4 (the homepage of REISE says it requires 1.5). I have not seen either problem you mentioned before..

Reply:
![image](https://user-images.githubusercontent.com/77214628/143322334-a6cbd94a-a9fb-4f5b-9f70-ed1f49866c8a.png)
![image](https://user-images.githubusercontent.com/77214628/143322346-95ea5537-f05e-4dae-8cbe-d1d60b743070.png)
For example I am printing one the DF field after updating and the output looks fine

Reply:
Okay, that must be the issue on my end. I'm using Julia 1.6 and haven't encountered this issue (or others) before. Thanks for verifying

Reply:
Actually I updated my Julia a couple of weeks ago and then I found that everything fails.. Then I was kinda forced to rollback to 1.5.

Reply:
@dongqi-wu and @lanesmith thanks for pointing this out, I've started an issue to coordinate information on the 1.5 vs. 1.6 issue and develop a plan of attack: https://github.com/Breakthrough-Energy/REISE.jl/issues/162